src/2024h2/std-verification.md

```diff
@@ -0,0 +1,147 @@
+
+First, we will instrument some unsafe functions of the forked Rust Standard Library with function contracts,
+and safe abstractions with safety type invariants.
+
+Then we will employ existing verification tools to verify that the annotated unsafe functions are in-fact safe as long
+as its contract pre-conditions are preserved. And we will also check that any post condition is respected.
+With that, we will work on proving that safe abstractions do not violate any safety contract, and that it does not
+leak any unsafe value through its public interface.
+
+Type invariants will be employed to verify that unsafe value encapsulation is strong enough to guarantee the safety
+of the type interface. Any method should be able to assume the type invariant, and it should also preserve the type
+invariant. Unsafe methods contract must be enough to guarantee that the type invariant is also preserved at the end
+of the call.
+
+Finally, we hope to contribute upstream contracts and type invariants added to this fork using the [experimental contract
+support proposed in this MCP](https://github.com/rust-lang/compiler-team/issues/759).
+
+This is open source and very much open to contributions of tools/techniques/solutions.
+We introduce problems (currently phrased as challenges) that we believe are important to the Rust and verification
+communities. These problems can be solved by anyone.
+
+### The "shiny future" we are working towards
+
+We are working towards the enhancement of Rust verification tools, so it can eventually be incorporated as part of
+regular Rust development cycle for code that require the usage of unsafe Rust.
+
+The Rust Standard Library is the perfect candidate given its blast radios and its extensive usage of unsafe Rust
+to provide performant abstractions.
+
+## Design axioms
+
+- **No runtime penalty**: Instrumentation must not affect the standard library runtime behavior, including performance.
+- **Automated Verification**: Our goal is to verify the standard library implementation. Given how quickly the standard 
+library code evolves, automated verification is needed to ensure new changes preserve the properties previously verified.
+- **Contract as code**: Keeping the contract language and specification as close as possible to Rust syntax and 
+semantics will lower the barrier for users to understand and be able to write their own contracts.
+
+## Ownership and team asks
+
+**Owner:** @celinval
+
+| Subgoal                            | Owner(s) or team(s)  | Notes                                                                 |
+|------------------------------------|----------------------|-----------------------------------------------------------------------|
+| Discussion and moral support       | ![Team][] [libs][]   |                                                                       |
+| Standard review                    | ![Team][] [libs][]   | We would like to contribute upstream the contracts added to the fork. |
```

Comment:
I don't think you plan to do this **in the next 6 months**, right? I would leave this out

Reply:
Ah, re-reading, maybe you do! I can see the logic in it, but it's worth discussing with the libs team a bit more.